### DISCUSSION TITLE
Is there a way to initialize in the air and cruise normally？

### DISCUSSION THREAD

**Discussion by cx-zzz** [JSBSim-Team/jsbsim#1192] [Q&A]
Hi, i am new to JSBSim and would like to know if it is possible to initialize A320 in JSBSim in the air like normally flying? 
I have see the issue [298](https://github.com/JSBSim-Team/jsbsim/issues/298) and tried to  change xml file, but the aircraft was just located at the position i set, but the parameter like throtle was just 0. 
Is there any way to achieve this？

**Comment by bcoconni** [comment]
Hi, the short answer is yes this is feasible. However could you please provide more details about what you are doing and how ?

**Reply by cx-zzz** [reply]
Hi, i am working on a flying game based on the UE demo. I want players to choose their own flying stages, such as takeoff and cruise, just like Microsoft Flight Simulator, but i don't know how to set JSBSim to a specific state like cruise.
Any help is appreciated - thank you!

**Comment by cx-zzz** [comment]
I attempted to save the tied_properties in the FGPropertyManager class, but when I overwrote tied_properties with the saved data, I encountered a JSBSim::TrimFailureException. I'm not sure if this method is correct, but it doesn't seem feasible at the moment.

**Reply by bcoconni** [reply]
You need to set the initial conditions and call `FGFDMExec::RunIC`.

Basically, you get a pointer to the FDM initial conditions by calling `FGFDMExec::GetIC`, you modify/update the initial conditions (for example by calling `FGInitialConditions::ResetIC`) and then call `FGFDMExec::RunIC`.
```c++
#include <FGFDMExec.h>

int main(int argc, char **argv)
{
  JSBSim::FGFDMExec FDMExec;
  auto ic = FDMExec.GetIC();
  ic->ResetIC( ... );
  FDMExec.RunIC();
  bool result = true;
  while (result) result = FDMExec.Run();
}
```
There are a number of different methods in `FGInitialConditions` to initialize your aircraft to your liking.

**Reply by bcoconni** [reply]
If you need to trim the aircraft, you must call `FGFDMExec::DoTrim()` after `FGFDMExec::RunIC()`.
For on ground trimming, call `FGFDMExec::DoTrim(tGround)`, for in-flight trimming call `FGFDMExec::DoTrim(tFull)`.

**Reply by cx-zzz** [reply]
Thank you！It works but i still got JSBSim::TrimFailureException when i tried to ` DoTrim(tFull)`. Is it a must or how can i fix it?

**Comment by gallonmate** [comment]
The UE5 demo already includes the option to set initial properties. Follow the red circles from top to bottom in the image below:
After placing an aircraft in the level. Select the aircraft blueprint in the World Outliner. Then in the Details panel below that, now select the JSBSimMovement component. After this you can scroll down and set different Initial Conditions. 

![image](https://github.com/user-attachments/assets/00ba1813-1a83-432b-85df-7ba851723fc1)

**Reply by cx-zzz** [reply]
Thank you for your reply, i will try this way later.
